### DISCUSSION TITLE
Question regarding the KeepAliveIntervalMs parameter

### DISCUSSION THREAD

**Discussion by tldQuantic** [microsoft/msquic#2706] [General]
Hi,
I have a question regarding how to detect all types of disconnection between client<->server.

There are especially two cases that I want to detect:
1- From the client point of view, I want to be notified when the server crash or is killed by the user.
2- From the server point of view, I want to be notified when the client crash or is killed by the user.

On client side, I added this parameter:
settings.KeepAliveIntervalMs = _uiKeepAliveIntervalMs;
settings.IsSet.KeepAliveIntervalMs = TRUE;

So, now, if the server crash, the client will be notified by MsQuic callback depending of the KeepAliveIntervalMs value and when the DisconnectTimeoutMs is reach.

The problem is that I can't do the same thing on server side. 

According to this documentation:
https://github.com/microsoft/msquic/blob/9d5574187b6d9b14ae3a47dba1cd5e7b1b5e2b27/docs/Deployment.md
_"They can be enabled on either the client or server side, **but only need to be enabled on one side**. They can be enabled either dynamically in the code or globally via the settings. To enable keep alives via the settings, set the KeepAliveIntervalMs setting to a reasonable value, such as 20000 (20 seconds)."_

If I try to enable this parameter on both side, this will have the same behaviour as the IdleTimeoutMs parameter and that is not what i want.

So, I am wondering if I missed something somewhere :)
Any help will be appreciated !

**Comment by nibanks** [comment]
>If I try to enable this parameter on both side, this will have the same behaviour as the IdleTimeoutMs parameter and that is not what i want.

Why are you saying you get the same behavior as IdleTimeoutMs in this case? You should just have keep alives polling on both sides.

**Reply by tldQuantic** [reply]
Hi,
I made the test again by enabling the KeepAliveIntervalMs parameters on both side.
I don't know what I did wrong previously but now everything is working fine !
Sorry for the inconvenience.
Thanks
